### DISCUSSION TITLE
Differential Version of `bitmap.convert(mi.Bitmap.PixelFormat.RGB, mi.Struct.Type.UInt8)` for `mi.TensorXf`

### DISCUSSION THREAD

**Discussion by jensenzhoujh** [mitsuba-renderer/mitsuba3#659] [Q&A] [answered]
Hi, 

I want to clip the rendered image from `mi.render()` to the range (0, 1) such that it can receive the gradient from off-the-shelf vision models with typical input range being (0, 1). But I noticed in mitsuba3 such transformation is only supported using `mi.util.convert_to_bitmap()` which is based on a non-differentiable Bitmap. I am wondering whether we have a counterpart directly for `mi.TensorXf` allowing me to do that. 

I've noticed in the tutorial `** (1.0 / 2.2)` is used to do that. Does this single approximated step serve the purpose?

Thanks

**Comment by njroussel** [comment, marked as answer]
Hi @shallowtoil 

The `Bitmap` class, indeed, doesn't have any sort of gradient tracking capabilities. Also, I don't believe we provide any built-in functions to do the tonemapping/sRGB ramp differentiably.
The snippet you mention from the tutorial can technically serve that purpose. Again, a 2.2 gamma correction is just an approximation.
